src/Event/Productions/ProductionCommandHandler.php

```diff
@@ -52,6 +61,45 @@ public function handleRemoveEventFromProduction(RemoveEventFromProduction $comma
     public function handleMergeProductions(MergeProductions $command): void
     {
         $toProduction = $this->productionRepository->find($command->getTo());
+        $fromProduction = $this->productionRepository->find($command->getFrom());
+
         $this->productionRepository->moveEvents($command->getFrom(), $toProduction);
+
+        $this->markNewMergedPairsAsLinked(
+            $fromProduction,
+            $toProduction
+        );
+    }
+
+    public function handleSkipEvents(SkipEvents $command): void
+    {
+        $this->similaritiesClient->excludePermanently(EventPair::fromArray($command->getEventIds()));
+    }
+
+    /** @param string $eventId
+     * @param ProductionId $productionId
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     * @todo: move logic to event/event handler
+     */
+    private function markAsLinked(string $eventId, ProductionId $productionId): void
+    {
+        try {
+            $pairs = $this->productionRepository->findEventPairs($eventId, $productionId);
+            $this->similaritiesClient->excludeTemporarily($pairs);
+        } catch (EntityNotFoundException $e) {
+        }
+    }
+
+    private function markNewMergedPairsAsLinked(
+        Production $from,
+        Production $to
+    ) {
+        $eventPairs = [];
+        foreach ($from->getEventIds() as $eventIdFrom) {
+            foreach ($to->getEventIds() as $eventIdTo) {
+                $eventPairs[] = new EventPair($eventIdFrom, $eventIdTo);
+            }
+        }
+        $this->similaritiesClient->excludeTemporarily($eventPairs);
```

Comment:
Suggestion (also sent you this on Slack but posting it here for reference):
Maybe we should move this to a decorator?